src/pkg/filters/filters.go

```diff
@@ -24,21 +27,48 @@ const (
 	Fails
 	// passthrough for the target
 	IdentityValue
-	// "foo" is a prefix of "foo/bar/baz"
+	// "foo" is a prefix of "foobarbaz"
 	TargetPrefixes
+	// "foo" equals any complete element prefix of "foo/bar/baz"
+	TargetPathPrefix
+	// "foo" equals any complete element in "foo/bar/baz"
+	TargetPathContains
 )
 
 func norm(s string) string {
 	return strings.ToLower(s)
 }
 
+// normPathElem ensures the string is:
+// 1. prefixed with a single os path delimiter (ex: `/`)
+// 2. suffixed with a single os path delimiter (ex: `/`)
+// This is done to facilitate future regex comparisons
+// without re-running the prefix-suffix addition multiple
+// times per target.
+func normPathElem(s string) string {
+	if len(s) == 0 {
+		return s
+	}
+
+	if s[0] != path.PathSeparator {
+		s = string(path.PathSeparator) + s
+	}
+
+	if s[len(s)-1] != path.PathSeparator {
+		s += string(path.PathSeparator)
+	}
```

Comment:
this may not always do the "right" thing. Things get tricky if the input has an escaped `/` at the end (i.e. `foo\/`). In that case, the last character is the path separator, but it's not acting as a path separator. However, just checking the last 2 characters is also insufficient because there could be any number of escaped `\` characters as in `foo\\\\\/`

It would probably be easiest to just do `s = path.TrimTrailingSlash(s)+path.PathSeparator`

Also, trailing `/` is currently being trimmed off by the CLI [here](https://github.com/alcionai/corso/blob/b5a274600818a075a48f862c7430769fcebad1fc/src/cli/utils/flags.go#L62) ([PR](https://github.com/alcionai/corso/pull/1175)). If we're always adding a `/` here we could probably ditch the CLI code (but do keep the CLI tests in `cli/backup/(onedrive|exchange)_test.go`)

Reply:
Good call, sticking with the trim-then-add.

Per the cli behavior, I'm fine keeping that in-tact.  Cleaning inputs is good for downstream handling, and filters will want to ensure things are normalized either way.

Reply:
heh yea, it was a bit of a trip figuring out all the edge cases for trim when writing the path package 🙃